### ISSUE TITLE
Rework logic to determine what moves are valid

### ISSUE THREAD

**Issue by kronmillerg** [CheeseLord/wat#25]
Currently doMove has:

```
    if (target && target.blocksMovement) {
        userError("Can't move there; something's in the way.");
        return;
    }
```

without regard for what else is on that cell. So the only thing preventing you from moving on top of another character is that the other character will have a higher Z-index than the ground, and therefore when you click on that square it'll target the character instead of the ground.

**Comment by kronmillerg** [comment]
This also means we're basically not checking validity of moves for AI characters, since they're not going to be bound by Z-indices. Basically they'll just artificially query for the ground at that spot, and then of course the checks will pass because they specifically requested the ground.

**Comment by kronmillerg** [comment]
This was incidentally fixed as part of 3219de3c7c2a89c21e334cb43bf31faf13f38bbc.
